### ISSUE TITLE
Error Column of Table db_version in mangosthree database-develop21

### ISSUE THREAD

**Issue by xgh23** [mangosthree/database#69]
if we execute all scripts, we will find that, some script files required db_version with column **from** "
World/Updates/Rel18/12335_01_creature_template_spells-patch.sql
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1054 (42S22) at line 1: Unknown column 'required_12335_01_creature_template_spells' in 'db_version'"
**To** "World/Updates/Rel18/12752_01_mangos_reputation_spillover_template.sql
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1054 (42S22) at line 1: Unknown column 'required_12751_01_mangos_phase' in 'db_version'"

I check the db_version create script in the branch "develop21", which name is "mangosdLoadDB.sql"：
CREATE TABLE `db_version` (
  `version` int(3) NOT NULL,
  `structure` int(3) NOT NULL,
  `content` int(3) NOT NULL,
  `description` varchar(30) NOT NULL DEFAULT '',
  `comment` varchar(150) DEFAULT '',
  PRIMARY KEY (`version`,`structure`,`content`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 ROW_FORMAT=COMPACT COMMENT='Used DB version notes';

However， in the main is ：
CREATE TABLE `db_version` (
  `version` varchar(120) DEFAULT NULL,
  `creature_ai_version` varchar(120) DEFAULT NULL,
  `cache_id` int(10) DEFAULT '0',
  `required_12752_01_mangos_reputation_spillover_template` bit(1) default NULL
) ENGINE=MyISAM DEFAULT CHARSET=utf8 ROW_FORMAT=FIXED COMMENT='Used DB version notes';

What I've done is just executed install_linux.sh on centos

**Comment by xieguohua** [comment]
[123111.txt](https://github.com/mangosthree/database/files/491632/123111.txt)

the Attach file is my installing log with all error info.
This may help for you guys.

**Comment by billy1arm** [comment]
@xieguohua - if you are attempting to install a fresh database using the develop21 branch, then you should install the main database, then only the updates from the Rel21 folder

If you are installing from a much older database, I would question why not start fresh as it's very badly out of date

**Comment by xieguohua** [comment]
@billy1arm Ok, I've installed from develop21 directly. 
So, I need to install mangosthree database, and then Rel21 in branch develop21?

**Comment by billy1arm** [comment]
@xieguohua - yes

**Comment by Warkdev** [comment]
Hello, can  you please give me the sequence in which you executed your scripts ?

**Comment by xieguohua** [comment]
@Warkdev Just edit install_linux.sh,  then execute "sh install_linux.sh"

**Comment by xieguohua** [comment]
@billy1arm Can'nt believable! 
/World/Updates/Rel21/Rel20_to_BaseRel21_Updates/

Here you refactoring the table db_version? (before this using db_version column name as db's version checking, but here you try to use column value).

Is this really? install with step :　1)install file mangosdCreateDB.sql、mangosdLoadDB.sql in trunk file world/setup by sequencial, and then install all scripts in world/setup/FullDB; 2) The same in trunk file Character/setup and Realmd/setup.  3)Then as your description, I need to import scripts of develop21/World/Updates/Rel21(Rel18 and Rel19 was included in trunk files).
But is This true, We changed all mechanism of db_version here.

**Comment by ASLAnthony** [comment]
@ShellGh - I inherited MangosThree in a mess, we are now working to make the Rel21 branch work the same as the other branches.
The db_version table was completely refactored some time ago to make the upgrade process much easier to manage,.

There are now 3 types of DB updates:
1) Major Release changes
2) Database structure changes
3) Database Content updates

1 & 2 will prevent the core starting is there is a mismatch
3 will display a warning during start-up, but the server will start

As a final note, 
If you are installing the develop21 branch and loaded the base database, is it not logical to assume that the database would also be develop21 ?
- Why would you attempt to install updates from an earlier release over the top of a newer database ?

As for the folder 'Rel20_to_BaseRel21_Updates' - As the readme clearly states, only apply these if you are upgrading from an earlier release.

**Comment by ASLAnthony** [comment]
@ShellGh - Just to clarify, are you trying to update an older database where you need to keep the custom data you have.

or 

Are you installing fresh ?

**Comment by billy1arm** [comment]
@ShellGh - I inherited MangosThree in a mess, we are now working to make the Rel21 branch work the same as the other branches.
 The db_version table was completely refactored some time ago to make the upgrade process much easier to manage,.

There are now 3 types of DB updates:
 1) Major Release changes
 2) Database structure changes
 3) Database Content updates

1 & 2 will prevent the core starting is there is a mismatch
 3 will display a warning during start-up, but the server will start

As a final note, 
 If you are installing the develop21 branch and loaded the base database, is it not logical to assume that the database would also be develop21 ?
•Why would you attempt to install updates from an earlier release over the top of a newer database ?

As for the folder 'Rel20_to_BaseRel21_Updates' - As the readme clearly states, only apply these if you are upgrading from an earlier release.

Just to clarify, 

are you trying to update an older database where you need to keep the custom data you have ?

or 

Are you installing fresh ?

**Comment by ShellGh** [comment]
@billy1arm I'm sorry I'm confuse now. 
You mean, I've no need to install database with branch develop21 even I've installed the code of server with branch develop21?

Then, why I've tried to install the Update scripts in  'Rel20_to_BaseRel21_Updates'? As described two days ago, your answer for :`Ok, I've installed from develop21 directly. So, I need to install mangosthree database, and then Rel21 in branch develop21?` But you see, folder 'Rel20_to_BaseRel21_Updates' was subfolder of "Rel21".

And you have some IM? I've several question on installing the server in several kind of platform, too(FreeBSD is OK, bug CentOS and Windows7 both failed). I can provide installing log.
